### PR TITLE
Splitting keypair class into secp and ed based keycurves

### PR DESCRIPTION
**Pull request by twtaylorbitgo**
From stellar seed still TBD

Ticket: BG-22623

### CODE REVIEW

File: src/coin/baseCoin/baseKeyPair.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import {AddressFormat} from "./enum";
```
Comment (gonzaalvarez70) [inline src/coin/baseCoin/baseKeyPair.ts:1]: I think that you don't have eslint enabled, it was configured within the pre-commit step so I'm not sure what happened. But I  just noticed that there are a few indentations/spacing issues.

File: src/coin/baseCoin/baseKeyPair.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import {KeyPairOptions} from "./iface";
+
+/**
+ * Base keys and address management.
+ */
+export abstract class BaseKeyPair {
```
Comment (argjv) [inline src/coin/baseCoin/baseKeyPair.ts:7]: This should be an interface

File: src/coin/baseCoin/ed25519KeyPair.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import {BaseKeyPair} from "./baseKeyPair";
+
+export abstract class Ed25519KeyPair extends BaseKeyPair {
```
Comment (argjv) [inline src/coin/baseCoin/ed25519KeyPair.ts:3]: Class JS doc

File: src/coin/baseCoin/baseKeyPair.ts (hunk 1 above)
Comment (twtaylorbitgo) [reply src/coin/baseCoin/baseKeyPair.ts:1]: oh that's funny. thanks

File: src/coin/baseCoin/baseKeyPair.ts (hunk 2 above)
Comment (twtaylorbitgo) [reply src/coin/baseCoin/baseKeyPair.ts:7]: I'll make it an interface for now

In terms of composition we need a base class for the KeyPair functionality though and should tack on any aspects of Keys via interfaces if this gets out of hand

Things like ```IFromXlmSeed```, etc.

**Pull request merged**